src/map/battle.cpp

```diff
@@ -689,6 +689,12 @@ int battle_calc_cardfix(int attack_type, struct block_list *src, struct block_li
 	t_race2 = status_get_race2(target);
 	s_defele = (tsd) ? (enum e_element)status_get_element(src) : ELE_NONE;
 
+	// When the attacker is a monster, then all bonuses on BF_WEAPON will work and no bonuses on BF_MAGIC
+	if (src->type == BL_MOB) {
+		flag |= BF_WEAPON;
+		flag &= ~BF_MAGIC;
```

Comment:
I know you probably have a reasoning, but it would be nice to also know about why you only change the flag and not the attack_type itself (maybe document that in an inline comment).

Reply:
I just tried to be minimal invasive. We don't know how the Aegis code works, but I know it only impacts the damage modifiers.